nipype/pipeline/engine/test_newengine/test_groupby.py

```diff
@@ -0,0 +1,188 @@
+from supernode import Node # or whatever
+
+import numpy as np
+
+import pytest, pdb
+from test_single_node import fun1, fun2, fun3
+from test_multi_nodes import fun4
+
+def fun2a(a, **dict):
+    pow = np.arange(3)
+    return a**pow
+
+
+@pytest.mark.parametrize("inputs_dict, expected_output", [ 
+        ({"a": [3, 4, 5]},
+         # output is 1dimensional, out[0] is related to a[0]=3
+         # a[:][1] is 0-dimensional, i.e. one element 
+         # there is no othe inputs states, so dictionary in a[:][1][0] is empty
+         [({"a":3}, ({}, 9)), ({"a":4}, ({}, 16)), ({"a":5}, ({}, 25))]), 
+        # do we want to allow 2D inputs a, when mapper="a"?
+        #({"a": [[3, 4, 5], [6, 7, 8]]}, [([3, 4, 5], [9, 16, 25]), ([6, 7, 8], [36, 49, 64])]),
+        ({"a": np.array([3, 4, 5])}, 
+         [({"a":3}, ({}, 9)), ({"a":4}, ({}, 16)), ({"a":5}, ({}, 25))]),
+        ])
+def test_singlenode_groupbyind_1(inputs_dict, expected_output):
+    # for now i'm doing only groupByIndex, should I keep the original reducer name?
+    N  = Node(inputs=inputs_dict, mapper="a", groupByInd="a", interface=fun1)
+    N.run()
+    
+    # should we have both N.output and N.output_grouped/reduced ??
+    for (i, out) in enumerate(N.output["out"]):
+        assert out[0] == expected_output[i][0] 
+        assert (out[1][1] == expected_output[i][1][1]).all() #still assuming that values will be returned as arrays 
+
+    
+@pytest.mark.parametrize("inputs_dict, expected_output", [
+        ({"a":[3, 1, 8], "b":[0, 1, 2]}, 
+         # output is 1dimensional, out[0] is for a[0]=3, etc.
+         # since mapper was a.b out[0][1] has only one element, for b[0]=0 
+         [({"a":3}, ({"b":0}, 0)), ({"a":1}, ({"b":1}, 1)), ({"a":8}, ({"b":2}, 16))]), # since there is a dot product, after grouping over "a" we have still only single values (but they are connected to values of "b") 
+        ])
+def test_single_groupbyind_2(inputs_dict, expected_output):
+    N = Node(interface=fun3, mapper=('a','b'), groupByInd="a")
```

Comment:
@satra, what would it be the coding-style guideline here for keyword arguments? camel-case like here for `groupByInd` or underscores like the `inputs_names` of the Function interface?